### ISSUE TITLE
train_op in linear regression

### ISSUE THREAD

**Issue by mzur** [astorfi/TensorFlow-World#12]
Is defining `train_op` [for each data point and epoch anew](https://github.com/astorfi/TensorFlow-World/blob/b60bddaa0b77fb9d407c1b980d6e1d4ba22e66d2/codes/2-basics_in_machine_learning/linear_regression/code/linear_regression.py#L85) really needed? I'm new to TensorFlow so I can't tell why or why not this would make sense. For me, the regression seems to work fine (and much faster) if the line is removed.

**Comment by astorfi** [comment]
Thank you so much for spending time and reading it. Basically, **train_op** is a crucial graph tensor which must be run by the TensorFlow session to perform a training step. As an example, performing a training step can be equivalent to updating the weights. In the code that you mentioned. *train_op* is the optimizer object, not just the loss.

You can refer to [TensorFlow Mechanics ](https://www.tensorflow.org/get_started/mnist/mechanics) for further details. Moreover, a detailed explanation is provided in [Convolutional Neural Networks](https://github.com/astorfi/TensorFlow-World/tree/master/docs/tutorials/3-neural_network/convolutiona_neural_network) documentation.

**Comment by mzur** [comment]
Thanks for the explanation.
